<?xml version="1.0"?>
<robot xmlns:xacro="http://wiki.ros.org/xacro"
       name="ur5" >

  <xacro:arg name="transmission_hw_interface" default="hardware_interface/EffortJointInterface"/>

  <!-- common stuff -->
  <xacro:include filename="$(find ur_description)/urdf/common.gazebo.xacro" />

  <!-- ur5 -->
  <xacro:include filename="$(find ur5_gripper_control)/urdf/ur5.urdf.xacro" />

  <!-- arm -->
  <xacro:ur5_robot prefix="" joint_limited="true"
    shoulder_pan_lower_limit="${-pi}" shoulder_pan_upper_limit="${pi}"
    shoulder_lift_lower_limit="${-pi}" shoulder_lift_upper_limit="${pi}"
    elbow_joint_lower_limit="${-pi}" elbow_joint_upper_limit="${pi}"
    wrist_1_lower_limit="${-pi}" wrist_1_upper_limit="${pi}"
    wrist_2_lower_limit="${-pi}" wrist_2_upper_limit="${pi}"
    wrist_3_lower_limit="${-pi}" wrist_3_upper_limit="${pi}"
    transmission_hw_interface="$(arg transmission_hw_interface)"
  />

  <!-- Realsense Camera -->
  <xacro:include filename="$(find ur5_gripper_control)/urdf/_d435.urdf.xacro" />
  <xacro:include filename="$(find ur5_gripper_control)/urdf/stereo_camera.urdf.xacro" />

  <xacro:sensor_d435 parent="world" name="camera_1_depth" topics_ns="camera_1_depth">
    <origin xyz="1.25 0 1.5" rpy="0 ${pi/6} ${pi}"/>
  </xacro:sensor_d435>

  <xacro:sensor_d435 parent="world" name="camera_2_depth" topics_ns="camera_2_depth">
    <origin xyz="-1.25 0 1.5" rpy="0 ${pi/6} 0.0"/>
  </xacro:sensor_d435>

  <xacro:stereo_camera_v0 name="camera_1_stereo" image_format="R8G8B8"
        parent="world" hfov="90" focal_length="639.997649"
        image_width="1280" image_height="720">
      <origin xyz="1.25 0 1.25" rpy="0 ${pi/6} ${pi}" />
  </xacro:stereo_camera_v0>

  <xacro:stereo_camera_v0 name="camera_2_stereo" image_format="R8G8B8"
      parent="world" hfov="90" focal_length="639.997649"
      image_width="1280" image_height="720">
      <origin xyz="-1.25 0 1.25" rpy="0 ${pi/6} 0.0" />
  </xacro:stereo_camera_v0>

  <link name="world" />

  <link name="pedestal">
    <inertial>
      <origin xyz="0.0 0.0 0.1" rpy="0.0 0.0 0.0"/>
      <mass value="0.0"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
    <visual name="">
      <origin xyz="0.0 0.0 0.1" rpy="0.0 0.0 0.0"/>
      <geometry>
        <cylinder radius="0.07" length="0.2"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <origin xyz="0.0 0.0 0.1" rpy="0.0 0.0 0.0"/>
      <geometry>
        <cylinder radius="0.07" length="0.2"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </collision>
  </link>

  <gazebo reference="pedestal">
    <selfCollide>true</selfCollide>
    <material>Gazebo/FlatBlack</material>
  </gazebo>

  <joint name="world_joint" type="fixed">
    <parent link="pedestal" />
    <child link = "base_link" />
    <origin xyz="0.0 0.0 0.225" rpy="0.0 0.0 0.0" />
  </joint>

  <joint name="world_pedestal" type="fixed">
    <parent link="world" />
    <child link = "pedestal" />
    <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
  </joint>

</robot>
